batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -55,178 +55,248 @@ pub async fn send_messages(
         );
 
         nonce += U256::one();
-
         let msg = ClientMessage::new(verification_data.clone(), wallet.clone()).await;
-        let msg_bin = cbor_serialize(&msg).map_err(SubmitError::SerializationError)?;
-        ws_write
-            .send(Message::Binary(msg_bin.clone()))
-            .await
-            .map_err(SubmitError::WebSocketConnectionError)?;
-
-        debug!("Message sent...");
-
-        let msg = match response_stream.next().await {
-            Some(Ok(msg)) => msg,
-            _ => {
-                return Err(SubmitError::GenericError(
-                    "Connection was closed without close message before receiving all messages"
-                        .to_string(),
-                ));
+        let msg_bin = match cbor_serialize(&msg) {
+            Ok(bin) => bin,
+            Err(e) => {
+                error!("Error while serializing message: {:?}", e);
+                sent_verification_data.push(Err(SubmitError::SerializationError(e)));
+                return sent_verification_data;
```

Comment:
For a later improvement (to not hold this one back): we might serialize first in a loop, fail the whole thing in that case, and then after all serialization succeeds we can try to send everything, possibly "waiting in parallel" as the async model promotes.